### PR TITLE
Add new alert rules for throttling

### PR DESCRIPTION
**Pull request by gabrielcocenza**
- If OpenSearch is throttling, this is an alert that optimizations are necessary like scaling the number of nodes or changing queries and indexing patterns

### CODE REVIEW

File: src/alert_rules/prometheus/prometheus_alerts.yaml [hunk 1]
```diff
@@ -105,3 +105,23 @@
     "for": "1m"
     "labels":
       "severity": "alert"
+
+  - "alert": "OpenSearchFewNodes"
```
Comment (zmraul) [inline src/alert_rules/prometheus/prometheus_alerts.yaml]: In my opinion this alert is not too relevant. High availability requirements should be explained on docs, or deployment requisites, not trigger once integrated with observability.

**Review by zmraul — DISMISSED** [review]

File: src/alert_rules/prometheus/prometheus_alerts.yaml (hunk 1 above)
Comment (gabrielcocenza) [reply src/alert_rules/prometheus/prometheus_alerts.yaml]: I kind agree. However, the generated metrics cannot ensure that a node is lost apart from checking if the `up`metric is working or not. I created this as a warning just to trigger something not critical to operators know that the current cluster is not operating as expected.

File: src/alert_rules/prometheus/prometheus_alerts.yaml [hunk 2]
```diff
@@ -105,3 +105,23 @@
...
+    "annotations":
+      "message": "Cluster {{ $labels.cluster }} has less nodes than expected. It's recommend to have at least 3 nodes for high availability."
+      "summary": "Few OpenSearch nodes on cluster"
+    "expr": |
+      sum by (cluster) (opensearch_cluster_nodes_number) / count by (cluster) (opensearch_cluster_nodes_number) < 3
```
Comment (phvalguima) [inline src/alert_rules/prometheus/prometheus_alerts.yaml]: We are supporting from 1+ nodes, so this alert would make sense if we have *at least 3x units* and then we have   one or more failing.

File: src/alert_rules/prometheus/prometheus_alerts.yaml [hunk 3]
```diff
@@ -105,3 +105,23 @@
...
+    "for": "5m"
+    "labels":
+      "severity": "warning"
+
+  - "alert": "OpenSearchThrottling"
+    "annotations":
+      "message": "Cluster {{ $labels.cluster }} is throttling. Please optimize queries and indexing patterns or consider scale the application."
```
Comment (phvalguima) [inline src/alert_rules/prometheus/prometheus_alerts.yaml:134]: That is a good one!

**Review by phvalguima — COMMENTED** [review]
Hey @gabrielcocenza we cannot unfortunately limit ourselves to 3+ nodes. The scenario of 1-node and 2-node cluster is also supposed to work. On the other side, the throttling alert is a really good one!

File: src/alert_rules/prometheus/prometheus_alerts.yaml (hunk 2 above)
Comment (gabrielcocenza) [reply src/alert_rules/prometheus/prometheus_alerts.yaml]: I had the assumption that for production OpenSearch we would have at least 3 nodes.
I'll remove this rule and just let the throttling one

File: src/alert_rules/prometheus/prometheus_alerts.yaml (hunk 1 above)
Comment (gabrielcocenza) [reply src/alert_rules/prometheus/prometheus_alerts.yaml]: I've removed the node count alert. We can add in the future if necessary

File: src/alert_rules/prometheus/prometheus_alerts.yaml (hunk 2 above)
Comment (gabrielcocenza) [reply src/alert_rules/prometheus/prometheus_alerts.yaml]: Rule removed

**Review by phvalguima — APPROVED** [review]

**Review by zmraul — APPROVED** [review]

**Pull request merged**